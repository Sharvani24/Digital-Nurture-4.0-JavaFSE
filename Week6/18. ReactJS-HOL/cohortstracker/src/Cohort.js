export const CohortData = [
  {
    code: 'CT101',
    name: 'Full Stack Dev',
    startDate: '2025-01-10'
  }
];
