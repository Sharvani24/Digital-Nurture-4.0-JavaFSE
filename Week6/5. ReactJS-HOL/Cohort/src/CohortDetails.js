// src/CohortDetails.js
import React from "react";
import styles from "./components/CohortDetails.module.css";

function CohortDetails({ cohort }) {
  const statusStyle = {
    color: cohort.status.toLowerCase() === "ongoing" ? "green" : "blue",
  };

  return (
    <div className={styles.box}>
      <h3 style={statusStyle}>{cohort.name}</h3>
      <dl>
        <dt>Started On</dt>
        <dd>{cohort.startDate}</dd>
        <dt>Current Status</dt>
        <dd>{cohort.status}</dd>
        <dt>Coach</dt>
        <dd>{cohort.coach}</dd>
        <dt>Trainer</dt>
        <dd>{cohort.trainer}</dd>
      </dl>
    </div>
  );
}

export default CohortDetails;