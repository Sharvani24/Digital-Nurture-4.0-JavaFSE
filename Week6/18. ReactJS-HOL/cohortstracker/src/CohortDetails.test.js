import React from 'react';
import { shallow, mount } from 'enzyme';
import CohortDetails from './CohortDetails';
import { CohortData } from './Cohort';

describe('CohortDetails Component Tests', () => {
  test('should render component without crashing', () => {
    const wrapper = shallow(<CohortDetails />);
    expect(wrapper.exists()).toBe(true);
  });

  test('should initialize props correctly', () => {
    const wrapper = mount(<CohortDetails cohort={CohortData[0]} />);
    expect(wrapper.props().cohort).toEqual(CohortData[0]);
  });

  test('should display cohort code in h3 tag', () => {
    const wrapper = mount(<CohortDetails cohort={CohortData[0]} />);
    expect(wrapper.find('h3').text()).toContain(CohortData[0].code);
  });

  test('should match snapshot', () => {
    const wrapper = shallow(<CohortDetails cohort={CohortData[0]} />);
    expect(wrapper).toMatchSnapshot();
  });
});
