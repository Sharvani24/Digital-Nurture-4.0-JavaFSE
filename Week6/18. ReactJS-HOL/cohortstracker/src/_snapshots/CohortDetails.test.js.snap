// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CohortDetails Component Tests should match snapshot 1`] = `
<div>
  <h3>
     -
    <span />
  </h3>
  <dl>
    <dt>
      Started On
    </dt>
    <dd>
      2025-01-10
    </dd>
    <dt>
      Current Status
    </dt>
    <dd />
    <dt>
      Coach
    </dt>
    <dd />
    <dt>
      Trainer
    </dt>
    <dd />
  </dl>
</div>
`;
